app/src/main/java/com/example/mbanking/details/DateBottomSheet.kt

```diff
@@ -0,0 +1,179 @@
+        }
+    ) {
+
+        Box(
+            modifier
+                .fillMaxSize()
+                .background(Color.Black)
+        ) {
+
+
+            Column(
+                modifier
+                    .fillMaxSize()
+                    .padding(20.dp)
+            ) {
+                Text(
+                    text = "Filter by date",
+                    color = Color.White,
+                    fontSize = 28.sp,
+                    fontWeight = FontWeight.Bold
+                )
+                Text(
+                    text = "Start date",
+                    color = Color.White,
+                    fontSize = 17.sp,
+                    modifier = modifier.padding(top = 30.dp),
+                    fontWeight = FontWeight.Light
+                )
+                OutlinedTextField(
+                    value = startDateSheet, onValueChange = { startDateSheet = it },
+                    modifier
+                        .fillMaxWidth()
+                        .padding(top = 7.dp),
+                    shape = RoundedCornerShape(10.dp),
+                    textStyle = TextStyle(Color.White),
+                    isError = checkDate(startDateSheet, validError),
+                    readOnly = true,
+                    trailingIcon = {
+                        Icon(painter = painterResource(id = R.drawable.calendar),
+                            contentDescription = null,
+                            modifier
+                                .clickable { startSheetState = true }
+                                .size(20.dp))
+                    }
+                )
+                Text(
+                    text = "End date",
+                    color = Color.White,
+                    fontSize = 17.sp,
+                    modifier = modifier.padding(top = 15.dp),
+                    fontWeight = FontWeight.Light
+                )
+                OutlinedTextField(
+                    value = endDateSheet,
+                    onValueChange = { endDateSheet = it },
+                    modifier
+                        .fillMaxWidth()
+                        .padding(top = 7.dp),
+                    shape = RoundedCornerShape(10.dp),
+                    textStyle = TextStyle(Color.White),
+                    isError = checkDate(endDateSheet, validError),
+                    readOnly = true,
+                    trailingIcon = {
+                        Icon(painter = painterResource(id = R.drawable.calendar),
+                            contentDescription = null,
+                            modifier
+                                .clickable { endSheetState = true }
+                                .size(20.dp))
+                    }
+                )
+                Button(
+                    onClick = {
+                        validError = if (startDateSheet != "" && endDateSheet != "") {
+                            scope.launch { modalBottomSheetState.hide() }.invokeOnCompletion {
```

Comment:
why did you use coroutines there? It's dangerous in compose

Reply:
done

Reply:
no, you still use coroutine (scope.launch{}). Can you tell me the reason why?